### ISSUE TITLE
Compile to byte-code

### ISSUE THREAD

**Issue by ccll** [PistonDevelopers/dyon#459]
Often developers don't want to distribute script source in plain text, so compiling to byte-code (or obfuscation) is very useful, is this possible in the future of dyon?

**Comment by bvssvni** [comment]
Yes, I believe this is a possibility. I wonder if I get time to work on it, though.

**Comment by bvssvni** [comment]
Dyon executes code from AST, so the code is very high level if we serialize the AST. Breaking changes to the AST will break the code.

An alternative is storing meta data from Piston-Meta. This will make it possible to load code faster.

**Comment by ccll** [comment]
I'm not a PL expert, but I guess it's easier to do obfuscation with the AST rather than compiling to byte-code?
For example renaming symbols to meaning-less `a/b/c/d`, like the uglifier for Javascript, I personally think this should suffice for a scripting language.

**Comment by bvssvni** [comment]
You could also use the meta syntax to rewrite the source code (using Piston-Meta to get the location of variables) and then inline the code with `include_str!`.

**Comment by ccll** [comment]
I'll check it out, Thanks!
